src/renderer/apis/settings.ts

```diff
@@ -18,7 +20,7 @@ type SettingsUpdate<T> =
  * Once the settings data has been copied into the `SettingsManager`, it can be read and written synchronously.
  * The `SettingsManager` automatically queues and dispatches updates to the file system in the background.
  */
-export class SettingsManager<T extends Record<string, unknown>> {
+export class SettingsManager<T extends Record<string, Jsonifiable>> {
```

Comment:
Can't you do away with the record and make the whole thing JSON?

Reply:
No, because settings are read/written like values in a map. The settings API imitates the methods of `Map`, but the type is defined using objects to allow different types for different keys. The top level of settings for a thing will always be a Record<string,Jsonifiable>, never an array or any other JSON value.